### PR TITLE
Add Typescript Typings

### PR DESCRIPTION
**Pull request by sebald**
Noticed that there are not typings yet :)

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+	/**
+	 * Whether to show cursor and allow navigation inside text input with arrow keys.
+	 */
+	showCursor?: boolean;
+
+	/**
+	 * Function to call when value updates.
+	 */
+	onChange: (value:string) => void;
```
Comment (vadimdemedes) [inline index.d.ts]: I'm not familiar with TS, but shouldn't there be a space before type?

File: index.test-d.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import * as React from 'react';
+import InkTextInput from '.';
+
+const handler = (value:string) => console.log(value);
+
+const Input = () => <InkTextInput value="foo" onChange={handler} />;
+const AllPropsInput = () =>
```
Comment (vadimdemedes) [inline index.test-d.tsx]: Could you wrap that JSX into `(` and `)`?

**Comment by vadimdemedes** [comment]
Thanks!

File: index.test-d.tsx (hunk 2 above)
Comment (sebald) [reply index.test-d.tsx]: done :)

**Comment by vadimdemedes** [comment]
As I mentioned, I haven't used TS myself before, could you check if this PR follows https://github.com/sindresorhus/typescript-definition-style-guide? One thing I noticed is that https://yarnpkg.com/en/package/tsd could be used instead of `tsc`.

**Comment by sebald** [comment]
@vadimdemedes I always check Sindre's style guide, especially if he is involved in the project. I hope I got everything right.

Regarding `tsd`: currently it is not possible to test `.tsx` files. See https://github.com/SamVerschueren/tsd/issues/15

If you have any other comments, let me know. I am happy to improve the PR 🙂

**Comment by vadimdemedes** [comment]
Awesome, thanks for this PR!

**Comment by vadimdemedes** [comment]
Oh, I forgot, could you indeed add it to `test` script?

**Comment by sebald** [comment]
@vadimdemedes Added the typecheck script to tests. Will do the same here: https://github.com/vadimdemedes/ink-select-input/pull/17

Thanks for `ink` 🙇‍♂️

**Comment by vadimdemedes** [comment]
Great work, thank you!

**Pull request merged**